### PR TITLE
Update createTheme to overwrite MQ based on breakpoints + update applyStyles to read theme from context

### PR DESCRIPTION
**Pull request by dgiraldo313**

### CODE REVIEW

File: packages/core/src/Badge/Badge.js [hunk 1]
```diff
@@ -71,8 +71,8 @@ const Badge = styled.div`
   text-transform: uppercase;
   letter-spacing: ${themeGet('letterSpacings.caps')};
   ${space} ${type} ${color};
-  ${applySizes(sizes)}
-  ${applyVariations('Badge')}
+  ${({ theme }) => applySizes(sizes, undefined, theme.mediaQueries)};
```
Comment (dgiraldo313) [inline packages/core/src/Badge/Badge.js:74]: This ensure that the sizes are true to the current theme in context and not the static sizes from the DS theme.

File: packages/core/src/utils/createTheme.js [hunk 2]
```diff
@@ -151,11 +151,14 @@ const createPalette = ({ palette = {}, ...theme }) => {
...
   const mergedTheme = deepmerge(defaultTheme, theme)
 
+  const mediaQueries = customBreakpoints
+    ? createMediaQueries(customBreakpoints)
+    : mergedTheme.mediaQueries
   return {
     ...mergedTheme,
     contrastRatio: mergedTheme.contrastRatio || 2.6,
     breakpoints: addAliases(customBreakpoints || mergedTheme.breakpoints),
-    mediaQueries: addAliases(mergedTheme.mediaQueries),
```
Comment (dgiraldo313) [inline packages/core/src/utils/createTheme.js:158]: With this previous logic, only the breakpoints were updated. This means that any DS component would adhere to those new breakpoints. However, the media queries we still the same as the default theme, so they new breakpoints do not work when accessing the diff media query inside a styled component.

**Review by sdalonzo — COMMENTED** [review]
overall LGTM. made a few minor suggestions in Slack. withholding approval for now, since we are going to publish a beta for testing first.

File: packages/core/src/utils/createMediaQueries.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Create media queries array based on the passed in breakpoints
+ *
+ * @param {Array} breakpoints The passed in breakpoints array
+ *
+ * @returns {Array} The generated media queries array
+ */
+const createMediaQueries = (breakpoints) =>
+  breakpoints.map((n) => `@media screen and (min-width:${n})`)
```
Comment (craigpalermo) [inline packages/core/src/utils/createMediaQueries.js:9]: Could you add a test or two for this new function?

**Review by craigpalermo — COMMENTED** [review]
LGTM pending the addition of tests for the updated `createMediaQueries` function.

**Review by gbellini911 — APPROVED** [review]

**Review by sdalonzo — APPROVED** [review]

**Review by craigpalermo — APPROVED** [review]

**Pull request merged**